src/OrbitGgp/ClientTest.cpp

```diff
@@ -46,13 +48,16 @@ TEST(OrbitGgpClient, GetInstancesAsyncWorking) {
   ASSERT_THAT(client, HasValue());
 
   bool callback_was_called = false;
-  client.value()->GetInstancesAsync(
-      [&callback_was_called](ErrorMessageOr<QVector<Instance>> instances) {
-        QCoreApplication::exit();
-        callback_was_called = true;
-        ASSERT_TRUE(instances.has_value());
-        EXPECT_EQ(instances.value().size(), 2);
-      });
+
+  auto future = client.value()->GetInstancesAsync();
+  orbit_base::ImmediateExecutor executor;
+  future.Then(&executor, [&callback_was_called](ErrorMessageOr<QVector<Instance>> instances) {
+    EXPECT_FALSE(callback_was_called);
+    callback_was_called = true;
+    ASSERT_THAT(instances, HasValue());
+    EXPECT_EQ(instances.value().size(), 2);
+    QCoreApplication::exit();
```

Comment:
I was just looking at the docs of `QCoreApplication::exit` and I don't see that this function is promised to be thread-safe. So that means you need to use the main thread executor here.